### PR TITLE
Return only our keys from the Cargo.toml parse.

### PR DESCRIPTION
**Pull request by daithiocrualaoich**

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -4,75 +4,114 @@ use std::fs::File;
...
+#[test]
+fn test_doc_commit_message_config() {
+    // Check doc_commit_message is set to "(cargo-gh-pages) Generate docs." in
+    // Cargo.toml of this repository.
+    let config: Result<Table, FatalError> = parse_config();
+    let table: Table = config.expect("Parse cargo file failed.");
+
+    assert_eq!(get_str(&table, "doc-commit-message"),
+               Some("(cargo-gh-pages) Generate docs."));
 }
```
Comment (passy) [inline src/config.rs]: Nice additional test coverage.

File: src/main.rs [hunk 2]
```diff
@@ -14,38 +15,34 @@ mod git;
...
+    let properties: Table = try!(config::parse_config());
 
-    // Verify the TOML configuration if present.
-    if let Some(invalid_keys) = config::verify_release_config(&cargo_file) {
-        for invalid_key in invalid_keys {
+    // Verify the TOML configuration.
+    let valid_keys = vec![config::SIGN_COMMIT,
+                          config::PUSH_REMOTE,
+                          config::DOC_BRANCH,
+                          config::DOC_COMMIT_MESSAGE];
```
Comment (passy) [inline src/main.rs]: Could this be a high-level const?

File: src/main.rs [hunk 3]
```diff
@@ -14,38 +15,34 @@ mod git;
...
     let sign = args.is_present("sign") ||
-               config::get_release_config(&cargo_file, config::SIGN_COMMIT)
-        .and_then(|f| f.as_bool())
-        .unwrap_or(false);
+               config::get_bool(&properties, config::SIGN_COMMIT).unwrap_or(false);
     let git_remote = args.value_of("push-remote")
-        .or_else(|| {
-            config::get_release_config(&cargo_file, config::PUSH_REMOTE).and_then(|f| f.as_str())
-        })
+        .or_else(|| config::get_str(&properties, config::PUSH_REMOTE))
```
Comment (passy) [inline src/main.rs]: I still find this a little funny to read, but that's just unfamiliarity. It's good that "lazy vals" are so easily created.

**Comment by passy** [comment]
Neat! :shipit:

File: src/main.rs (hunk 2 above)
Comment (daithiocrualaoich) [inline src/main.rs]: I tried that but vec was doing heap allocations somewhere. I know what you mean, though, I don't like this.

Good news is I woke up knowing what the problem is... we want there to be a struct in config.rs that wraps the Table and provides the constructor and accessors. Will be able to remove this validation check nicely into the constructor.r

**Comment by daithiocrualaoich** [comment]
Does the weird squirrel mean ship it? I read the alt text but that meme is way over my head. :)

**Pull request merged**